src/components/modals/ModalUserEdit.tsx

```diff
@@ -0,0 +1,363 @@
+import { Box, Button, Grid, Modal, SimpleGrid, Space, Text, TextInput, Title } from "@mantine/core";
+import React, { useEffect, useRef, useState } from "react";
+import { useTranslation } from "react-i18next";
+import SortableTree from "react-sortable-tree";
+import FileExplorerTheme from "react-sortable-tree-theme-file-explorer";
+import { Mail, User } from "tabler-icons-react";
+
+import { signup } from "../../actions/authActions";
+import { fetchDirectoryTree, manageUpdateUser, updateUserAndScan } from "../../actions/utilActions";
+import { useAppDispatch, useAppSelector } from "../../store/store";
+import { PasswordEntry } from "../settings/PasswordEntry";
+
+type Props = {
+  isOpen: boolean;
+  updateAndScan?: boolean;
+  userToEdit: any;
+  selectedNodeId?: string;
+  onRequestClose: () => void;
+  userList: any;
+  createNew: boolean;
+  firstTimeSetup?: boolean;
+};
+
+export function ModalUserEdit(props: Props) {
```

Comment:
More of a general question, ss there a specific reason why we're not using `@mantine/form` here?

Reply:
Would be nice, but most of the state is from the legacy component. So this would be an optional fix from my point of view.